File: tts_server/api/system.py
from fastapi import APIRouter, Depends, HTTPException, status

from tts_server.auth.dependencies import get_current_user_id
from tts_server.schemas import DeviceInfoOut
from tts_server.worker.qwen3_worker import get_worker

router = APIRouter(prefix="/system", tags=["system"])


@router.get("/device", response_model=DeviceInfoOut)
def device_info(_user_id: str = Depends(get_current_user_id)) -> DeviceInfoOut:
    worker = get_worker()
    info = worker.info
    if not info:
        raise HTTPException(status_code=status.HTTP_503_SERVICE_UNAVAILABLE, detail="Worker not initialized")
    return DeviceInfoOut(
        device=info.device,
        model_id=info.model_id,
        xpu_gate_passed=info.xpu_gate_passed,
        xpu_gate_reason=info.xpu_gate_reason,
        xpu_memory_bytes=info.xpu_memory_bytes,
        cpu_warmup_seconds=info.cpu_warmup_seconds,
        xpu_warmup_seconds=info.xpu_warmup_seconds,
    )

File: clients/android/tts-client/src/main/kotlin/it/polimar/tts/client/model/System.kt
package it.polimar.tts.client.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class HealthResponse(
    val status: String,
    @SerialName("worker_ready")
    val workerReady: Boolean,
)

@Serializable
data class DeviceInfo(
    val device: String,
    @SerialName("model_id")
    val modelId: String,
    @SerialName("xpu_gate_passed")
    val xpuGatePassed: Boolean,
    @SerialName("xpu_gate_reason")
    val xpuGateReason: String,
    @SerialName("xpu_memory_bytes")
    val xpuMemoryBytes: Long,
    @SerialName("cpu_warmup_seconds")
    val cpuWarmupSeconds: Float,
    @SerialName("xpu_warmup_seconds")
    val xpuWarmupSeconds: Float? = null,
)
